.circleci/config.yml

```diff
@@ -4,37 +4,55 @@
 version: 2
 jobs:
   build:
+    working_directory: ~/java
+    parallelism: 1
     docker:
-      # specify the version you desire here
-      - image: circleci/openjdk:8-jdk
-
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/postgres:9.4
-
-    working_directory: ~/repo
+    - image: circleci/openjdk:8
 
     environment:
       # Customize the JVM maximum heap limit
       MAVEN_OPTS: -Xmx3200m
 
     steps:
+      - run: sudo apt-get update && sudo apt-get install -y python-pip && sudo pip install awscli --upgrade
+
       - checkout
 
-      # Download and cache dependencies
       - restore_cache:
-          keys:
-          - v1-dependencies-{{ checksum "pom.xml" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
+          key: v1-java-maven-{{ checksum "pom.xml" }}
 
-      - run: mvn dependency:go-offline
+      - run:
+          name: Compile the code
+          command: mvn compile
+
+      - run:
+          name: Run tests
+          command: mvn test
 
       - save_cache:
+          key: v1-java-maven-{{ checksum "pom.xml" }}
           paths:
-            - ~/.m2
-          key: v1-dependencies-{{ checksum "pom.xml" }}
+          - ~/.m2
+
+      - store_test_results:
+          path: ~/java/target/surefire-reports
+
+      - run: mvn package
+
+      - deploy:
```

Comment:
Shouldn’t these be deploy jobs, called from the workflow, instead of deploy steps that are part of the build job?

Reply:
This has been done. Many bothans died to bring us this configuration.